balancer-js/src/modules/data/fee-distributor/repository.ts

```diff
@@ -100,4 +111,28 @@ export class FeeDistributorRepository implements BaseFeeDistributor {
 
     return Math.floor(midnight.getTime() / 1000) - daysSinceThursday * 86400;
   }
+
+  async getClaimableBalances(
+    userAddress: string,
+    claimableTokens: string[]
+  ): Promise<TokenBalance> {
+    const amounts = await this.feeDistributor.callStatic.claimTokens(
+      userAddress,
+      claimableTokens
+    );
+    return claimableTokens.reduce(
+      (tokens: { [key: string]: number }, token, index) => {
+        tokens[token] = amounts[index].toNumber();
```

Comment:
This is throwing: 
```
$ npx ts-node -P tsconfig.testing.json -r tsconfig-paths/register ./examples/claimVeBalRewards.ts
/Users/jg/Documents/balancer-sdk/balancer-js/node_modules/@ethersproject/logger/src.ts/index.ts:269
        const error: any = new Error(message);
                           ^
Error: overflow [ See: https://links.ethers.org/v5-errors-NUMERIC_FAULT-overflow ] (fault="overflow", operation="toNumber", value="51270986292799141088", code=NUMERIC_FAULT, version=bignumber/5.7.0)
    at Logger.makeError (/Users/jg/Documents/balancer-sdk/balancer-js/node_modules/@ethersproject/logger/src.ts/index.ts:269:28)
    at Logger.throwError (/Users/jg/Documents/balancer-sdk/balancer-js/node_modules/@ethersproject/logger/src.ts/index.ts:281:20)
    at throwFault (/Users/jg/Documents/balancer-sdk/balancer-js/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts:356:19)
    at BigNumber.toNumber (/Users/jg/Documents/balancer-sdk/balancer-js/node_modules/@ethersproject/bignumber/src.ts/bignumber.ts:186:13)
    at /Users/jg/Documents/balancer-sdk/balancer-js/src/modules/data/fee-distributor/repository.ts:125:40
    at Array.reduce (<anonymous>)
    at FeeDistributorRepository.getClaimableBalances (/Users/jg/Documents/balancer-sdk/balancer-js/src/modules/data/fee-distributor/repository.ts:123:28)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async /Users/jg/Documents/balancer-sdk/balancer-js/examples/claimVeBalRewards.ts:33:19 {
  reason: 'overflow',
  code: 'NUMERIC_FAULT',
  fault: 'overflow',
  operation: 'toNumber',
  value: '51270986292799141088'
}
```

Reply:
I will check

Reply:
I tried on my machine, and it works fine:

```
balancer-js git:(develop) npx ts-node -P tsconfig.testing.json -r tsconfig-paths/register ./examples/claimVeBalRewards.ts
┌────────────────────────────────────────────┬──────────────────┐
│                  (index)                   │      Values      │
├────────────────────────────────────────────┼──────────────────┤
│ 0x7B50775383d3D6f0215A8F290f2C9e2eEBBEceb2 │        0         │
│ 0xA13a9247ea42D743238089903570127DdA72fE44 │ 2304141496137171 │
│ 0xba100000625a3754423978a60c9317c58a424e3D │ 1870979002020152 │
└────────────────────────────────────────────┴──────────────────┘

    to:                  0xD3cf852898b21fc233251427c2DC93d3d604F3BB - BalancerMinter Address 0xD3cf852898b21fc233251427c2DC93d3d604F3BB
    from:                0x549c660ce2B988F588769d6AD87BE801695b2be3 - User Address 0x549c660ce2B988F588769d6AD87BE801695b2be3
    tokensOut:           0x7B50775383d3D6f0215A8F290f2C9e2eEBBEceb2,0xA13a9247ea42D743238089903570127DdA72fE44,0xba100000625a3754423978a60c9317c58a424e3D
    expectedTokensValue: 0,2304141496137171,1870979002020152
    functionName:        claimTokens
    callData:            0x88720467...24e3d
  


finally:
  
  const tx = { to: data.to', data: callData };
  const receipt = await (await signer.sendTransaction(tx)).wait();
```

from what we can see, the culprit is `this.feeDistributor.callStatic.claimTokens(userAddress,  claimableTokens)`, throwing the overflow error. 

What implementation of `Contract` are you using? 

In our example, the expected result is

```
[
  BigNumber { value: "0" },
  BigNumber { value: "2304141496137171" },
  BigNumber { value: "1870979002020152" }
]
```

No value of `51270986292799141088` is returned.  It seems that the fee-distributor contract is returning some weird result.

We are adding a check to intercept errors thrown by the contract, but this should be investigated at the chain level.
We also added a test to detect spurious values (null or undefined).